### PR TITLE
feat(pds-ember): add Loading component

### PR DESCRIPTION
**Pull request by CITguy**
Port `<Loading>` and its subcomponents from HCP, with the following differences:

1. some contextual styles previously present in HCP are not included
    - `.pdsAppContent > .fullHeight .loadingState` (no context to know if it's necessary)

JIRA:
  * PDS-161
  * PDS-162
  * PDS-163

### CODE REVIEW

File: packages/pds-ember/public/structure-icons/loading.css [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+@media screen and (-ms-high-contrast: active), (-ms-high-contrast: none) {
+  .structure-icon-loading-progress {
+    stroke-dasharray: none;
+    stroke-dashoffset: 0;
+  }
+
+  .structure-icon-loading {
+    animation: structure-icon-loading-simple-spin 1s infinite linear;
+  }
+}
```
Comment (CITguy) [inline packages/pds-ember/public/structure-icons/loading.css:56]: copied/vendored in from `@hashicorp/structure-icons`

File: packages/pds-ember/addon/components/pds/loading/elapsed/index.js [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+//       to handle calculation and formatting
+function timeSince(startTime) {
+  let now = new Date().getTime()
+  let diff = now - startTime
+  // TODO: consider using date-fns (or JS:Temporal) to calculate this
+  let hours = Math.floor((diff % (1000 * 60 * 60 * 24)) / (1000 * 60 * 60));
+  let minutes = Math.floor((diff % (1000 * 60 * 60)) / (1000 * 60));
+  let seconds = Math.floor((diff % (1000 * 60)) / 1000);
+  // TODO: consider using date-fns (or JS:Temporal) to format the duration
+  hours = hours < 10 ? '0' + hours : hours;
```
Comment (cbfx) [inline packages/pds-ember/addon/components/pds/loading/elapsed/index.js:24]: do you want to remove the semi-colons from here?

File: packages/pds-ember/addon/components/pds/loading/elapsed/index.js [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+    return setTimeout(() => {
+      run(() => {
+        this.elapsed = this.getElapsed()
+        this.timeout = this.poll()
+      })
+    }, 1000)
+  }
+
+  get startTime() {
+    return this.args.startTime || this.renderTime;
```
Comment (cbfx) [inline packages/pds-ember/addon/components/pds/loading/elapsed/index.js:72]: here's another semi-colon

**Review by cbfx — APPROVED** [review]
another great component. i'm curious about all of the lock changes. also, there are some semi-colons that you'd probably want to remove before merging. as far as the fullHeight thing goes, i think it's safe to exclude it and we'll use it localized in our app. nice work.

**Comment by CITguy** [comment]
@cbfx : lock file changes were a result of [updated dependencies](https://github.com/hashicorp/structure/pull/27/files#diff-902df337673f93e14982b6fad83cf51cf735fa8d78d54b5f2c18540dd851fc6d)

File: packages/pds-ember/addon/components/pds/loading/elapsed/index.js (hunk 2 above)
Comment (CITguy) [reply packages/pds-ember/addon/components/pds/loading/elapsed/index.js:24]: I'll update. Thanks for pointing these out.

**Pull request merged**